docs/glossary.md

```diff
@@ -11,3 +11,19 @@
 - **Game**: (AKA Game XML) refers to a specific XML within a map
 - **Map**: collection of files, eg: images, polygons text, properties, and at least one game XML
 
+## Battle Terms:
+- **Roll**: How many dice can the unit roll.
+- **Strength**: The number on the dice that determines whether a hit was made. Any number less than or equal to the number is a hit.
+- **Round**: A battle round includes both sides firing at each other. A battle can consist of multiple rounds. During and at the end of each round, casualties are removed from the game and can not participate anymore.
+- **Phase**: A battle phase is subpart of a battle round where specific rules are in play that determine what the units can do. Some phases allow the units to retreat. Other phases allow the units to fire at each other.
+- **General Phase**: This phase is the default phase for units to fire. The attacker fires first and then the defender retaliates. Any casualties that the attacker hits can retaliate. At the end of the phase, all casualties are removed from the game or transformed into other units which do not participate in the battle.
+- **First Strike Phase**: This phase is for units to fire and remove casualties before the General Phase. Historically, this was used for submarines to sink sea based units as a "sneak attack". Now, any unit can participate during this phase if they have the `isFirstStrike` attribute. Units that fire during this phase will not participate during the General Phase. At the end of the phase, all casualties are removed from the game or transformed into other units which do not participate in the battle.
+  - If an opposing side has an `isDestroyer` unit, this phase is skipped.
+  - WW2V2 always has this phase. The casualties will not be removed at the end of the phase if an `isDestroyer` unit is present on the opposing side.
+  - `Defending Subs Sneak Attack` game property controls whether the defender gets to fire in the phase.
+- **Targeted Attack Phase**: This phase is one of the first battle phases. Units determine their strength and roll based on their AA attributes (such as `attackAA` and `offensiveAttackAA`). If a unit has AA attributes, they can fire at this time. The `typeAA` and `targetsAA` attributes determine what units are targeted during this attack. At the end of the phase, all casualties are removed from the game or transformed into other units which do not participate in the battle. Units that fire during this phase can fire during later phases if they have normal strength and roll attributes.
+- **Retreat Phase**: This phase allows the attacker to retreat from battle. All units retreat together.
+  - Amphibious assaults only allow retreating of planes if `Attacker Retreat Planes` property is set or land based attackers and airplanes if `Partial Amphibious Retreat` is set.
+- **Can Evade Phase**: This phase allows units with `canEvade` to retreat or submerge (determined by `Submersible Subs` property).
```

Comment:
Would 'Can evade phase' be better named the 'evasion phase'? 
Do we expose "can evade" as a label to players? I think "submerge subs" is the term most players would recognize. For other maps with non-sub units having 'canEvade', then calling it the 'evasion phase' would make sense (in this case we would have aliases, one is just more specific if you only have subs with 'canEvade').

Reply:
Is this glossary for developers (including map makers) and players?  I was under the impression this is more for developers.

I agree about showing the name "submerge/retreat subs" or "evade(?) units" depending on the situation.

Reply:
Submerged units usually don't retreat. Also we have rulesets where the decision to submerge is made before rolling dice - and not during the Press Attack/Retreat step of a battle.
So this is not necessarily one and the same phase.

Reply:
I've renamed this phase to "Evasion Phase" and added a similar term "Submerge Subs Phase" as another name for this phase.

> Submerged units usually don't retreat.
If `Submersible Subs` is not set, then submarines can retreat during this phase.  That's what I was trying to say with "retreat or submerge (determined by `Submersible Subs` property)".  I'm open on suggestions to how to reword that to be more clear.

> Also we have rulesets where the decision to submerge is made before rolling dice - and not during the Press Attack/Retreat step of a battle.
There is a line below this that says "`Sub Retreat Before Battle` defines if this phase occurs before the First Strike Phase or at the end of the Battle Round."  Is that clear enough?

Reply:
Sounds good to me.